src/v/raft/persisted_stm.cc

```diff
@@ -119,8 +120,20 @@ file_backed_stm_snapshot::load_snapshot() {
     }
 
     storage::snapshot_reader& reader = *maybe_reader;
-    iobuf meta_buf = co_await reader.read_metadata();
-    iobuf_parser meta_parser(std::move(meta_buf));
+    auto meta_fut = co_await ss::coroutine::as_future(reader.read_metadata());
```

Comment:
This is only one place that can throw (though perhaps the most likely): shouldn't this protection extend to the entire portion of the method after the reader is opened, perhaps by pulling all of it into its own method and so that the single future returned by that method can have a `finally -> close()` attached?

Reply:
fixed